quality_control_oca/models/product_product.py

```diff
@@ -16,3 +18,62 @@ class ProductProduct(models.Model):
         inverse_name="product",
         string="Quality control triggers",
     )
+    created_inspections = fields.Integer(
+        compute="_compute_count_inspections", string="Created inspections"
+    )
+    done_inspections = fields.Integer(
+        compute="_compute_count_inspections", string="Done inspections"
+    )
+    passed_inspections = fields.Integer(
+        compute="_compute_count_inspections", string="Inspections OK"
+    )
+    failed_inspections = fields.Integer(
+        compute="_compute_count_inspections", string="Inspections failed"
+    )
+
+    def _compute_count_inspections(self):
+        data = (
+            self.env["qc.inspection"]
+            .sudo()
+            .read_group(
+                [("product_id", "in", self.ids)],
+                ["product_id", "state"],
+                ["product_id", "state"],
+                lazy=False,
+            )
+        )
+        product_data = {}
+        for d in data:
+            product_data.setdefault(d["product_id"][0], {}).setdefault(d["state"], 0)
+            product_data[d["product_id"][0]][d["state"]] += d["__count"]
+        for product in self:
+            count_data = product_data.get(product.id, {})
+            product.created_inspections = sum(count_data.values())
+            product.passed_inspections = count_data.get("success", 0)
+            product.failed_inspections = count_data.get("failed", 0)
+            product.done_inspections = (
+                product.passed_inspections + product.failed_inspections
+            )
+
+    def button_qc_inspection_per_product(self):
+        return self._action_qc_inspection_per_product()
+
+    def _action_qc_inspection_per_product(self):
+        product_ids = self.ids
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "quality_control_oca.action_qc_inspection"
+        )
+        domain = [("product_id", "in", product_ids)]
+        qc_type = self.env.context.get("qc_type", "all")
+        if qc_type == "done":
+            domain.append(("state", "not in", ["draft", "waiting"]))
+        elif qc_type == "passed":
+            domain.append(("state", "=", "success"))
+        elif qc_type == "failed":
+            domain.append(("state", "=", "failed"))
+        action["domain"] = domain
+        if len(product_ids) == 1:
+            context = ast.literal_eval(action.get("context"))
```

Comment:
Please don't do literal eval. That's very counter-productive. `context` should be already a dictionary.

Reply:
In my case it's not:

![imagen](https://github.com/OCA/manufacture/assets/12749832/bc349208-4e2d-4386-8401-9554c3827acb)

Am I doing something wrong?

Reply:
The problem is that `literal_eval` will crash on a lot of cases. Isn't another way to do this? What about having an action including already such context?

Reply:
@pedrobaeza done, move to XML actions, so python action coding (and `ast.literal_eval`) is no longer required at this point. I've lost `default_object_id` management, but I think it's not a key feature.